### PR TITLE
Increase length of the email and username

### LINKED ISSUE #20020: DB schema: varying(64) is not enough for email fields
### Apache Airflow version

2.2.2 (latest released)

### Operating System

composer-1.17.3-airflow-2.1.2

### Versions of Apache Airflow Providers

_No response_

### Deployment

Composer

### Deployment details

_No response_

### What happened
Originally reported here: 
https://stackoverflow.com/questions/69718827/receiving-http-401-when-accessing-cloud-composers-airflow-rest-api

Getting "401 Unauthorized" error when accessing Airflow Stable REST api with service accounts having email longer than 64 symbols.
Successfully Accessing API with service accounts email <64 symbols. 
Details:
https://stackoverflow.com/a/70217282/9583820

### What you expected to happen

Google Cloud Platform service accounts emails are usually quite long due to big domain part.
I would expect longer emails to work with Airflow

### How to reproduce

I think the same will reproduce for every environment, but my issue happened on this:
composer-1.17.3-airflow-2.1.2 (Google Cloud Platform)

Set "api-auth_backend" to "airflow.composer.api.backend.composer_auth"
Make sure your service account has required permissions (Composer User role should be sufficient)
Make sure your service account email length is >64 symbols
Provide Bearer token for the request

Result:
Requests to Stable API will fail with "401 Unauthorized" 

Change service account to have email <64 symbols
Result:
Requests sucessed

### Anything else

I think updating tables:
public.ab_register_user
public.ab_user

to have email field to be VARCHAR(320) as per 
RFC 5321 (the current SMTP spec, obsoletes RFC2821) states:
The maximum total length of a user name or other local-part is 64 octets. The maximum total length of a domain name or number is 255 octets

Should fix the issue.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by lwyszomi**
When we using service account as a `email` or `username` we can reach a character limit of these fields.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
LGTM but we need another committer for that one

**Comment by uranusjr** [comment]
Instead of changing an arbitrary length to another, it may be better to use `ID_LEN` in `airflow.models.base`.

**Comment by lwyszomi** [comment]
@uranusjr `ID_LEN` for me should be used only for id fields to keep nameing convention clear.

File: airflow/migrations/versions/5e3ec427fdd3_increase_length_of_email_and_username.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+def upgrade():
+    """Increase length of email from 64 to 256 characters"""
+    with op.batch_alter_table('ab_user') as batch_op:
+        batch_op.alter_column('username', type_=sa.String(256))
+        batch_op.alter_column('email', type_=sa.String(256))
+
+
+def downgrade():
+    """Revert length of email from 256 to 64 characters"""
+    with op.batch_alter_table('user') as batch_op:
```
Comment (swalkowski) [inline airflow/migrations/versions/5e3ec427fdd3_increase_length_of_email_and_username.py]: Should be `ab_user` here as well?

Comment (lwyszomi) [reply airflow/migrations/versions/5e3ec427fdd3_increase_length_of_email_and_username.py]: @swalkowski 🦅 👁️

File: docs/apache-airflow/migrations-ref.rst [hunk 2]
```diff
@@ -23,7 +23,9 @@ Here's the list of all the Database Migrations that are executed via when you ru
 +--------------------------------+------------------+-----------------+---------------------------------------------------------------------------------------+
 | Revision ID                    | Revises ID       | Airflow Version | Description                                                                           |
 +--------------------------------+------------------+-----------------+---------------------------------------------------------------------------------------+
-| ``be2bfac3da23`` (head)        | ``7b2661a43ba3`` | ``2.2.3``       | Add has_import_errors column to DagModel                                              |
+| ``5e3ec427fdd3`` (head)        | ``be2bfac3da23`` | ``2.2.3``       | Increase length of email and username in ``ab_user`` table to ``256`` characters      |
```
Comment (swalkowski) [inline docs/apache-airflow/migrations-ref.rst]: Shouldn't we include changing column lengths in `ab_register_user` in this migration as well, given that you update `ab_register_user` table in `models.py`?

Comment (lwyszomi) [reply docs/apache-airflow/migrations-ref.rst]: 👍

**Review by potiuk — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
Only internittent errors that are being fixed already. Merging

**Pull request merged**